lib/rubygems/ext/ext_conf_builder.rb

```diff
@@ -40,6 +40,7 @@ def self.build(extension, dest_path, results, args = [], lib_dir = nil, extensio
       end
 
       ENV["DESTDIR"] = nil
+      ENV["MAKEFLAGS"] ||= "-j"
```

Comment:
Can we use `-j#{Etc.nprocessors + 1}` or something instead of unlimited jobs?
If a host doesn't have enough resources (CPU cores/memory/...), `-j` may be slower than no `-j`.